Doc/library/io.rst

```diff
@@ -911,21 +923,22 @@ Text I/O
 
    .. method:: seek(offset, whence=SEEK_SET, /)
 
-      Change the stream position to the given *offset*.  Behaviour depends on
-      the *whence* parameter.  The default value for *whence* is
-      :data:`SEEK_SET`.
-
-      * :data:`SEEK_SET` or ``0``: seek from the start of the stream
-        (the default); *offset* must either be a number returned by
-        :meth:`TextIOBase.tell`, or zero.  Any other *offset* value
-        produces undefined behaviour.
-      * :data:`SEEK_CUR` or ``1``: "seek" to the current position;
-        *offset* must be zero, which is a no-operation (all other values
-        are unsupported).
-      * :data:`SEEK_END` or ``2``: seek to the end of the stream;
-        *offset* must be zero (all other values are unsupported).
-
-      Return the new absolute position as an opaque number.
+      Change the stream position to the given byte *offset*,
```

Comment:
The existing docs/docstrings use offset, cookie, and pos. I don't understand what's meant by cookie. I understand what's meant by offset or position. If there is no need to use the same name, there is no need for this PR.

Reply:
I think *offset* is fine, as it is still an offset into the file even if not an integer byte offset. I agree we should stay away from *cookie* et al.

Reply:
"Cookie" means an opaque value. No, it is not an offset. It is several values (position of the buffer in the file, position in the buffer, state of the encoder and decoder, packed in a 257-bit integer. See code of `_pack_cookie()` and `_unpack_cookie()` and very complex code of `tell()` and `seek()` in `_pyio.py`.

This PR is needed because semantic of `tell()` and `seek()` in binary and text files is so different.

Reply:
Ouch, yes TextIOWrapper is a different thing. Very unfortunate.